### PR TITLE
[BugFix] Fix seeded random sampling with encoder-decoder models

### PR DESCRIPTION
**Pull request by njhill**
Currently providing the `seed` parameter with encoder-decoder models including the llama-3.2-vision ones will crash the server.

Planning to follow up with some better test coverage and hopefully some changes to make the implementation less fragile, but would be good to get this fix in quickly.

Thanks to @tjohnson31415 for helping find this!

```
ERROR 09-26 18:32:49 engine.py:130]   File "/opt/vllm/lib64/python3.11/site-packages/vllm/worker/enc_dec_model_runner.py", line 223, in execute_model
ERROR 09-26 18:32:49 engine.py:130]     output: SamplerOutput = self.model.sample(
ERROR 09-26 18:32:49 engine.py:130]                             ^^^^^^^^^^^^^^^^^^
ERROR 09-26 18:32:49 engine.py:130]   File "/opt/vllm/lib64/python3.11/site-packages/vllm/model_executor/models/mllama.py", line 939, in sample
ERROR 09-26 18:32:49 engine.py:130]     next_tokens = self.sampler(logits, sampling_metadata)
ERROR 09-26 18:32:49 engine.py:130]                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 09-26 18:32:49 engine.py:130]   File "/opt/vllm/lib64/python3.11/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
ERROR 09-26 18:32:49 engine.py:130]     return self._call_impl(*args, **kwargs)
ERROR 09-26 18:32:49 engine.py:130]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 09-26 18:32:49 engine.py:130]   File "/opt/vllm/lib64/python3.11/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
ERROR 09-26 18:32:49 engine.py:130]     return forward_call(*args, **kwargs)
ERROR 09-26 18:32:49 engine.py:130]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 09-26 18:32:49 engine.py:130]   File "/opt/vllm/lib64/python3.11/site-packages/vllm/model_executor/layers/sampler.py", line 273, in forward
ERROR 09-26 18:32:49 engine.py:130]     maybe_deferred_sample_results, maybe_sampled_tokens_tensor = _sample(
ERROR 09-26 18:32:49 engine.py:130]                                                                  ^^^^^^^^
ERROR 09-26 18:32:49 engine.py:130]   File "/opt/vllm/lib64/python3.11/site-packages/vllm/model_executor/layers/sampler.py", line 878, in _sample
ERROR 09-26 18:32:49 engine.py:130]     return _sample_with_torch(
ERROR 09-26 18:32:49 engine.py:130]            ^^^^^^^^^^^^^^^^^^^
ERROR 09-26 18:32:49 engine.py:130]   File "/opt/vllm/lib64/python3.11/site-packages/vllm/model_executor/layers/sampler.py", line 818, in _sample_with_torch
ERROR 09-26 18:32:49 engine.py:130]     multinomial_samples[sampling_type] = _multinomial(
ERROR 09-26 18:32:49 engine.py:130]                                          ^^^^^^^^^^^^^
ERROR 09-26 18:32:49 engine.py:130]   File "/opt/vllm/lib64/python3.11/site-packages/vllm/model_executor/layers/sampler.py", line 626, in _multinomial
ERROR 09-26 18:32:49 engine.py:130]     assert seq_group.generator is not None
ERROR 09-26 18:32:49 engine.py:130]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 09-26 18:32:49 engine.py:130] AssertionError
```

### CODE REVIEW

**Comment by Pernekhan** [comment]
LGTM!

**Review by ywang96 — APPROVED** [review]
LGTM! Thanks for the fix!

**Comment by nikola-rahman** [comment]
When will this bugfix be on pypi?

**Comment by nikola-rahman** [comment]
When can we expect this change to be on the pip repository? If my eyes and my cache aren't fooling me it still isn't.
